### PR TITLE
only enable amplifier when needed

### PR DESCRIPTION
**Pull request by jochenberger**
depends on #22

This should save some power and suppress some noise from the mp3 chip.
The downside is a slight popping sound when the AMP is toggled. I wonder if we can get rid of that too.

### CODE REVIEW

File: src/tonuino.cpp [hunk 1]
```diff
@@ -191,9 +193,28 @@ void Tonuino::disableStandbyTimer() {
   }
 }
 
+void Tonuino::enableAmplifier() {
+#if defined ALLinONE || defined ALLinONE_Plus
+  // if (!amplifierEnabled){
+    digitalWrite(ampEnablePin, getLevel(ampEnablePinType, level::active));
+    amplifierEnabled = true;
```
Comment (boerge1) [inline src/tonuino.cpp:200]: You should remove this line and the variable at all. It increases only flash and RAM usage.

**Review by boerge1 — CHANGES_REQUESTED** [review]

File: src/tonuino.cpp (hunk 1 above)
Comment (jochenberger) [reply src/tonuino.cpp:200]: Okay, I wasn't so sure about the effects of writing the same value to the pin twice.

File: src/tonuino.cpp [hunk 2]
```diff
@@ -50,6 +50,8 @@ void Tonuino::setup() {
   }
 
   SM_tonuino::start();
+  enableAmplifier();
```
Comment (boerge1) [inline src/tonuino.cpp:53]: This is not necessary. The state machine handles this.

**Comment by jochenberger** [comment]
It just occured to me that we probably have to do something when shortcuts are played in Idle state.

**Comment by jochenberger** [comment]
Or can we do this based on events from the mp3 player?

**Comment by boerge1** [comment]
Sorry for taking so long to investigate this change. 

I checked the current consumption of only the amplifier when the DFMiniMp3 player is playing nothing. It is about 13 mA. 
In normal Play mode the whole TonUINO needs 130 mA. 

Normally, in all the cases you want to switch out the amplifier the TonUINO starts the standby timer and after 5 minutes the TonUINO switches out. So, normally you have switched out the amplifier for max 5 minutes. 

If using your implementation we save max. 13 mA * 5 mins. As the TonUINO needs 130 mA when normal playing we win only a half minute normal playing. 

On the other side it is very difficult to find all the places in the code where the amplifier has to switched on for all the beeps. So I don't see the benefit of this change at all.

**Comment by jochenberger** [comment]
Very well.
I actually had two motivations for the PR.
The main one was that I heard noises in Idle or Standby mode which I suspected would go away if the amp is off. And I haven't heard them again with this change.
The other one was that I remember having read that this might be a possibile way to save some power. But I guess it wasn't thoroughly investigated back then.
Thanks for the numbers! I'll close the PR for now.

**Comment by jochenberger** [comment]
This was the discussion btw.
https://discourse.voss.earth/t/energetische-sanierung-tonuino-stromverbrauch-halbiert/9676/17

**Pull request closed without merging**